### DISCUSSION TITLE
Welcome to MSI-Dragon-Center-for-Linux Discussions!

### DISCUSSION THREAD

**Discussion by YoCodingMonster** [YoCodingMonster/OpenFreezeCenter#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by pratheepmm** [comment]
hey, I'm using MSI GF63 , I was a fan of linux but due to cooler boost and battery limit options which cannot be used in linux , I was worried about that,  and didn't migrate till now . finally I found your repository It's a really good thing . Thanks a lot . I don't know whether it will work for my machine or not but I was happy towards your approach. I think adding MSI battery limiter option in your application would be more better .

**Reply by YoCodingMonster** [reply]
Thanks for appriciation.

Thisapp will work.on your laptop. Just install the latest version amd follow the readme file.

And you will have a working fan control software with temp.and speeds monitering!!

**Reply by pratheepmm** [reply]
Hey ! thanks a lot bro . It is awesome .

**Reply by Datrix01** [reply]
@pratheepmm  bro can you tell me how to do it and is it working now in 2024 . I have same laptop, I have installed fedora40 on it and I am not getting how to I do it. I followed everything and this is what happening when I run ./install.sh
![Screenshot from 2024-09-10 15-42-34](https://github.com/user-attachments/assets/4a4960cb-f078-4a4a-b084-2c947c246eab)

**Comment by zbz95** [comment]
Hi! I wanted to say that you are doing great job, thank you very much!

As a suggestion, in dragon center there is an option to switch gears between "eco", "comfort", "sport" and "turbo". Will you add this option in future versions? I think this is the last feature that is lacking.

Thanks again!

**Reply by YoCodingMonster** [reply]
Surely I will add those Performance profiles..Without that the openFreezeCenter is incomplete 🤖

**Comment by mdpoleto** [comment]
Hi! First of all, thank you very much for your wonderful effort of developing this. I was planning to buy a MSI GL65 Leopard 10SEK-022 and after I saw your work here, I am more inclined to buy it. Have you seen tests with this model? I presume it should work, but I wanted to double check first.

**Reply by YoCodingMonster** [reply]
I haven't tested it on Your model...

I just have MSI GS65 on which I develope...and some of my friends have other models too. But the bottleneck here is that I need other people to test this application.

This should work fine and will not harn any MSI laptop in any ways. But to find a community of testers with MSI laptops is hard .. se ecially when everyone is gaming and no one or most of them don't want to contribute.

There are a couple of very helpful contributers to this project and I am bery thankful to them

But the limitation is not just the hardware . The infinite universe of linux distris make I hard for me to develope for each and every distro...

I am currently working on Ubuntu, Mint, Kali, Arch. And any varients of them!!!!

Your laptop will be safe 100% without any problem if you uae ununtu or any distro based on it

**Reply by mdpoleto** [reply]
@YoCodingMonster Thank you! I will finish up the purchase and I intend to use Ubuntu. I will let you know how your software works on it.

Also, I do know some python but I am not an expert on bash or C to help developing the code. But if I can assist in any form, just let me know.

**Reply by mdpoleto** [reply]
@YoCodingMonster Just tested on my GL65 and it is working just fine. The only thing I had to do is chmod the .sh files used during installation.

Also, every time I switch functions, it keeps asking me my sudo password. Seems there is still some permission minor bug to be sorted.

Nevertheless, thanks!

**Comment by Special-Niewbie** [comment]
@YoCodingMonster , man this app become really something you are really the best..... later, when you will be more quite, if you want I will give you some other idea if you are interesting. 

And I will be waiting the next releases with dual GPU Graphics, and then from my side to help you, I have one expensive spear SSD where I can test for you to install different Distros with the App (let me know if there is some particular Distro to test the App), where you can add for sure few MSI models to your Readme file list. Try to let know your app around, as much as you can, with the wish to receive some Donation from the users because you deserve it !!!

Thanks Infinitely for that...

**Comment by manchitro** [comment]
Hi, I'm currently using this on a GF63 9sc running Pop!_OS. it works fine, provided that I run it correctly. I was wondering if there is any way automate this? Can I setup so that run this script at startup without typing in my password everytime to run? I am using linux for the first time so I am a complete noob

**Reply by YoCodingMonster** [reply]
we are working on automating this software, will be working with the next release soon

**Comment by KolbjornBlix** [comment]
Thanks for the work you're doing! saved me from reinstalling Windows to adjust charging levels. Works like a charm on my GP65 Leopard 9SD. Using Zorin OS 16 Pro.

![GP65-Leopard-9SD](https://user-images.githubusercontent.com/79469515/130552700-0cf207d5-597d-456a-a69a-686809e7974e.png)

**Comment by buhaytza2005** [comment]
Works fine on MS-70 Stealth running Ubuntu. More details below:


Laptop:
```Version: E1771IMS.50E
	Version: REV:1.0
	Version: REV:0.B
	Version: N/A
	Version: Intel(R) Core(TM) i7-4700HQ CPU @ 2.40GHz
```
OS:
```NAME="Ubuntu"
VERSION="20.04.3 LTS (Focal Fossa)"
Kernel:
5.11.0-37-generic #41~20.04.2-Ubuntu
```

**Comment by leopseft** [comment]
Hello friend, thanks for your good work. Could you add support for the non gaming models too? Im a user of an MSI Modern 15 A11M-004 (11th gen of i7-1165G7). It uses the MSI Center Pro not the Dragon. I believe that it has fan schemes not so "aggressive" as the Gaming (Dragon) Models. Tell me what information do you need.

![εικόνα](https://user-images.githubusercontent.com/48049281/136427169-89c89840-e86a-4845-9422-085bc508fc7b.png)

![εικόνα](https://user-images.githubusercontent.com/48049281/136427285-afe4ddfc-0002-4721-b448-6d14c687e3e9.png)

The main idea is the same.

**Comment by YoCodingMonster** [comment]
working on MSI center!!

**Reply by leopseft** [reply]
I noticed that fans work ok on arch linux and the battery setting is saved (probably on uefi). But of course you cant make any changes from linux, only from Windows. So we are waiting for the MSI Pro Center!!

Thank you again!!

**Comment by ghost** [comment]
I recently installed Linux on a second NVME drive (but kept windows installed on the other NVME drive) on my MSI Delta 15 A5EFK.
I installed Garuda Dr4g0nized Edition.
It seems like OpenFreezeCenter is not doing anything for me.
The little penguin loads at the top where the taskbar is, but clicking on the various options results in nothing happening. 
Hopefully it can be fixed soon!
If you need more details, feel free to ask me for any information!

**Comment by YoCodingMonster** [comment]
i am preety new to git, i am working on it, since its final year of my college there has been a lot of work and i am grinding through to it all. Working is slow right now, but will pickup pace!, thanks for supporting and being helpful!!!!

**Comment by mohanvaddi** [comment]
cooler booster working well on MSI Modern 15, with POp!_OS. but unable change though the fan speeds.

``` 
Processor: Ryzen 5 5500U
```

**Comment by r4ph4r0ck5** [comment]
Hey folks.... seems a long time since someone talk here...

just to report that it is working on a MSI GS75 9SE

Only missing the third fan on the monitor screen but nothing that important

Fan curves working perfectly and I have no ideia about CPU profiles.... any way to check that?

Best,
Raphael

**Reply by ObsidianStack** [reply]
What distro????

**Comment by peetfrost** [comment]
Hi everyone,

at first I wanted to thank YoCodingMonster for his awesome job. I've purchased my MSI GP66 Leopard in summer and wanted to use Ubuntu but the fan has driven me crazy. It was spinning around 2500 rpm all the time and MSI Drageon Center gave me the opportunity to adjust this. I come from windows, but diving into machine / deep learning is pain in the *** on this os. So I decided to give Pop OS a try and this is pure bliss. However, I forgot about the fans and as I was googling around how to solve this, trying around 2 days with different solutions, I came across the OpenFreezeCenter. Thank god! There were only a few issues and maybe someone can help me here, since I am a unix noob.

I followed the steps in the instruction on a fresh installed and updated/upgraded Pop OS and with chmod I was able to run the startup script the first time. However, after running it a second time, nothing really showed up. I first had to run install_deps.sh and now it works. I can adjust fan speed, battery and so on. 

Whats left is, that under "advanced", temperature categories are not showing up, everywhere is just a "0°C". When I look at "monitoring", fan speed is shown around 4500 for the cpu, although its not spinning at all. Secure Boot is disabled and I tried enabling "Intel 11th gen" but nothing changed.

Clicking on the icon doesn't work everytime, even if I click exactly in the center of the icon, only in 3/5 clicks the menu dropdown menu pops out. 

Any suggestions on my minor issues here? Keep going on this, this is great!

**Comment by SelHaESGI** [comment]
Hey @YoCodingMonster , your software works perfectly on my GF75 Thin 9SC, thanks a lot :+1:

**Comment by r18duarte** [comment]
Can confirm it is working on my GS65 9SD with PopOS 21.10 on intel, still haven't tested with nvidia gpu, didn't bring the charger with me, will test when i get home and stress test it a bit. Cooler boost mode worked so i'm assuming fan control is 100% working

**Comment by ghost** [comment]
I own this model:
https://www.msi.com/Laptop/Delta-15-A5EX
The application doesn't seem to work at all on my model. None of the buttons do anything. 
I am using ArchLinux. 
Linux 5.16.9-arch1-1 x86_64

**Comment by khoihuynh2409** [comment]
Awesome, this app is the solution I'm looking for, it works on Model MSI Bravo 15 A4DCR
Distro Linux I use for testing: Ubuntu Mate, PopOS, Xubuntu, Arch Linux. And I don't forget to thank you so much

**Reply by r18duarte** [reply]
how can you get it working with arch? i'm not able to get it running on arch

**Comment by nzMike** [comment]
Hi
Currently trying to get this to work on GS63VR but keep getting 
bash: ./at_startup.sh: Permission denied
no matter what I change in the bios. any help would be greatly appreciated.
I'm running Solus Budgie, this is the first Linux distro (I'm new to linux) I have tried on this laptop, so that could also be the issue

**Reply by favallone20** [reply]
Did you try to give the permissions with chmod?

**Comment by Carbocode** [comment]
Works fine on my laptop, my only question is: does it will be supported in the next future? I'm asking this just because I saw the message on the code page but no other news

**Comment by Eltonno-Diabolo** [comment]
Thank you for your work.
I can confirm this works on MSI GP76 Leopard 10UG-291.

**Comment by favallone20** [comment]
Hi, I would like to use your program with my GF63 MSI. I know that it isn't supported, but I tried to launch the .sh file and I don't have any feedback from the terminal, is it normal? Or should something happen? Thank you in advance.

**Comment by EvanDSanders** [comment]
Are the numbers in the config.py file equivalent to those in MSI Center, or are they something else? 
```python
PROFILE = 3
    ## These look to be relative to those of MSI Center. Are they being percentages from 0 to 150%?
AUTO_SPEED = [[0, 40, 48, 56, 64, 74, 84], [0, 49, 58, 67, 76, 85, 85]] 
ADV_SPEED = [[0, 40, 48, 56, 64, 72, 80], [0, 48, 56, 64, 72, 79, 86]]
BASIC_OFFSET = 0 # Always added to the fan speed?
CPU = 1 # 10th gen or above, or is it CPU count?
AUTO_ADV_VALUES = [212, 13, 141] # I assume the larger numbers are for the CPU and GPU fans, respectively. 
# If that's the case, what's the 13 for?
COOLER_BOOSTER_OFF_ON_VALUES = [152, 2, 130] # Same as above but for the 2.
```
![image](https://github.com/YoCodingMonster/OpenFreezeCenter/assets/127012288/0ae1b68f-8671-4bd5-85bc-fc30f0f2afe9)
I want to build my own API using the Lite version of the script to ramp the curves myself and also make a KDE Plasma widget for the two fans in my laptop; what are the highest and lowest numbers I should put into this? If it works the way I think it does it would be 0 to 150%, is that right? 
My laptop is an MSI Raider GE66 12UHS, running Kubuntu 23.10.
